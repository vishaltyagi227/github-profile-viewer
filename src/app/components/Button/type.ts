import { ReactChild, ReactNode } from 'react';

export interface IButton {
  children: ReactChild;
  variant?: 'text' | 'outlined' | 'contained';
  type?: 'button' | 'reset' | 'submit';
  fullWidth?: boolean;
  disabled?: boolean;
  onClick?: () => void;
  startIcon?: ReactNode;
  className?: string;
}
