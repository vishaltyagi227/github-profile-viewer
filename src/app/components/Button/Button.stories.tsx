import React from 'react';
import { Story, Meta } from '@storybook/react';
import Button from '@components/Button';
import { IButton } from '@components/Button/type';

export default {
  title: 'Components/Button',
  component: Button,
} as Meta;

const Template: Story<IButton> = ({ ...args }) => <Button {...args} />;

export const Basic = Template.bind({});

Basic.args = {
  variant: 'contained',
  type: 'button',
  fullWidth: false,
  disabled: false,
  children: 'Button',
};
