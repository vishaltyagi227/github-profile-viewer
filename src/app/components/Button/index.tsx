import React from 'react';
import { Button } from '@mui/material';

import { VARIANT } from '@constant/index';
import useStyles from './style';
import { IButton } from './type';

const CustomButton = ({
  children,
  variant,
  type,
  fullWidth,
  disabled,
  onClick,
  startIcon,
  className,
}: IButton) => {
  const classes = useStyles();

  return (
    <Button
      variant={variant}
      type={type}
      startIcon={startIcon}
      fullWidth={fullWidth}
      className={className || classes.btn}
      disabled={disabled}
      onClick={onClick}
    >
      {children}
    </Button>
  );
};

CustomButton.defaultProps = {
  variant: VARIANT.text,
  fullWidth: false,
  disabled: false,
  onClick: () => {},
};

export default CustomButton;
