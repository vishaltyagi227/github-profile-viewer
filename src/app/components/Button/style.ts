import { createStyles, makeStyles } from '@mui/styles';
import { Theme } from '@mui/material/styles';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    btn: {
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      color: theme.customPallete.secondary.main,
      backgroundColor: theme.customPallete.primary.main,
      marginTop: theme.spacing(5),
      paddingBlock: theme.spacing(1),
      '&:hover': {
        backgroundColor: theme.customPallete.secondary.light,
      },
    },
  })
);

export default useStyles;
