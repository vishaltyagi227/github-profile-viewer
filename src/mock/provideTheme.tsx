import React, { ReactElement } from 'react';
import { ThemeProvider } from '@mui/material/styles';
import theme from '@theme/index';

export default (ui: ReactElement): ReactElement => (
  <ThemeProvider theme={theme}>{ui}</ThemeProvider>
);
