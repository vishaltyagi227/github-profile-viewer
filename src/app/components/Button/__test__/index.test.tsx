import React from 'react';
import { cleanup, render, RenderResult } from '@testing-library/react';
import renderer from 'react-test-renderer';
import provideTheme from '@mock/provideTheme';
import Button from '../index';

const renderingText = 'Test Button';
describe('<Button />', () => {
  let container: RenderResult;

  afterEach(cleanup);

  it('Should render correctly', () => {
    container = render(provideTheme(<Button>{renderingText}</Button>));
    const button = container.getByText(renderingText);
    expect(button?.textContent).toBe(renderingText);
  });

  it('Should be disabled when button set to disabled', () => {
    container = render(provideTheme(<Button disabled>{renderingText}</Button>));
    const button = container.getByText(renderingText);
    expect(button).toBeDisabled();
  });
});

describe('<Button /> Snapshot', () => {
  afterEach(cleanup);

  it('Should match with Snapshot when button is not disabled', () => {
    const tree = renderer
      .create(provideTheme(<Button>{renderingText}</Button>))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should match with Snapshot when button is disabled', () => {
    const tree = renderer
      .create(provideTheme(<Button disabled>{renderingText}</Button>))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
