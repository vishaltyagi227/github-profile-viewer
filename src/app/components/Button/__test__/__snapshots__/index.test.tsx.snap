// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Snapshot Should match with Snapshot when button is disabled 1`] = `
<button
  className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root Mui-disabled makeStyles-btn-3 css-1e6y48t-MuiButtonBase-root-MuiButton-root"
  disabled={true}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={-1}
  type="button"
>
  Test Button
</button>
`;

exports[`<Button /> Snapshot Should match with Snapshot when button is not disabled 1`] = `
<button
  className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root makeStyles-btn-3 css-1e6y48t-MuiButtonBase-root-MuiButton-root"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  Test Button
</button>
`;
